addons/goggles/functions/fnc_getExplosionIndex.sqf

```diff
@@ -13,13 +13,8 @@
  *
  * Public: No
  */
-private ["_effectIndex"];
-
-_effectIndex = switch true do {
-       case (_this <= 0.04): {0};
-       case (_this <= 0.06): {1};
-       case (_this <= 0.09): {2};
-       default {3};
-};
-
-_effectIndex
+#include "script_component.hpp"
+if (_this <= 0.04) exitWith {0};
+if (_this <= 0.06)) exitWith {1};
+if (_this <= 0.09) exitWith {2};
+3
```

Comment:
This whole file can be replaced with:

`floor (linearConversion [0, 0.1, _this, 0, 3, true]);`